### DISCUSSION TITLE
Workflow not running on push

### DISCUSSION THREAD

**Discussion by mgwc** [community/community#24943] [Actions]
<p>I’m setting up two workflows. One is supposed to run when I push to any branch other than main <em>and</em> the pushed code modifies something in the frontend subdirectory. Similarly, the other is supposed to run when I push to any branch other than main <em>and</em> the code modifies something in the backend subdirectory.</p>
<p>The .yml file for the frontend workflow looks like this:</p>
<pre><code class="lang-auto">name: frontend-push

# Controls when the action will run. 
on:
  # Triggers the workflow on pushes to any branch
  push:
    branches: 
      - '!main'
    paths:
      - 'frontend/**'
  
  # Allows you to run this workflow manually from the Actions tab
  workflow_dispatch:

# A workflow run is made up of one or more jobs that can run sequentially or in parallel
jobs:
  # This workflow contains two jobs: one called build and one called test
  build:
    # The type of runner that the job will run on
    runs-on: ubuntu-latest
    
    # Sets default working directory for all steps in job
    defaults:
      run:
        working-directory: frontend
    
    # Test multiple versions of node
    strategy:
      matrix:
        node-version: [12.x]
    

    # Steps represent a sequence of tasks that will be executed as part of the job
    steps:
      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
      - uses: actions/checkout@v2
      # Configure the specified Node.js version on the runner
      - name: Use Node.js ${{ matrix.node-version }}
        uses: actions/setup-node@v1
        with:
          node-version: ${{ matrix.node-version }}
      - run: ls -la
      - run: echo $PWD
      - run: npm install   # install the package and packages it depends on
      - run: npm run build --if-present   # creates a production build
      - run: npm test
</code></pre>
<p>I’ve tried testing this by creating a .txt file in the frontend directory and pushing the new file to a branch called test_github_actions, but the push did not trigger my workflow. I’ve been iterating on my workflow file for a while, and I’ve found and corrected (I think) a couple of errors, but I’ve gotten stuck here.</p>
<p>Is there an obvious issue with my .yml file?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>I think you’re just excluding <code>main</code>, without specifying that the workflow should run on other branches. Try this:</p>
<pre><code class="lang-yaml">on:
  push:
    branches: 
      - '**'
      - '!main'
</code></pre>

**Comment by mgwc** [comment]
<p>It looks like you were right! Thank you!</p>
